src/org/testKitGen/MkGen.java

```diff
@@ -96,7 +96,21 @@ private void writeSingleTest(List<String> testsInPlaylist, TestInfo testInfo, Fi
 				f.write(testTargetName + ": TEST_GROUP=" + testInfo.getLevelStr() + "\n");
 				f.write(testTargetName + ": TEST_ITERATIONS=" + testInfo.getIterations() + "\n");
 				f.write(testTargetName + ": AOT_ITERATIONS=" + testInfo.getAotIterations() + "\n");
-
+				
+				// Set special openjdk problem list for JVM options that contains FIPS profile
+				// This feature is ignored if TEST_FLAG contains FIPS
+				if (arg.getBuildList().contains("openjdk")) {
+					if (!arg.getTestFlag().contains("FIPS") && !var.getJvmOptions().isEmpty() && var.getJvmOptions().contains("-Dsemeru.customprofile=")) {
+						String[] splited = var.getJvmOptions().split("\\s+");
+						for (int i=0; i < splited.length; i++) {
+							if (splited[i].contains("-Dsemeru.customprofile=")) {
+								String fipsProfile = splited[i].replace("-Dsemeru.customprofile=", "").trim();
+								f.write(testTargetName + ": FIPS_VARIATION_PROBLEM_LIST_FILE=-exclude:$(Q)$(JTREG_JDK_TEST_DIR)/ProblemList-" + fipsProfile + ".txt$(Q)\n");
```

Comment:
Should this `break`, or continue searching for other `splited[i]`?

Reply:
We should break. Ideally, we only have one customprofile.